src/transformers/modeling_flax_utils.py

```diff
@@ -820,31 +886,27 @@ def from_pretrained(
                 _commit_hash=commit_hash,
             )
 
+        safetensors_from_pt = False
+        if filename == SAFE_WEIGHTS_NAME:
+            with safe_open(resolved_archive_file, framework="flax") as f:
+                safetensors_metadata = f.metadata()
+            if safetensors_metadata is None or safetensors_metadata.get("format") not in ["pt", "tf", "flax"]:
+                raise OSError(
+                    f"The safetensors archive passed at {resolved_archive_file} does not contain the valid metadata."
+                    " Make sure you save your model with the `save_pretrained` method."
+                )
+            safetensors_from_pt = safetensors_metadata.get("format") == "pt"
+
         # init random models
         model = cls(config, *model_args, _do_init=_do_init, **model_kwargs)
 
-        if from_pt:
+        if from_pt or safetensors_from_pt:
```

Comment:
Exposing my lack of knowledge about safe tensors here: if `safetensors_from_pt` is `True` here then is the reason we do `load_pytorch_checkpoint_in_flax_state_dict(model, resolved_archive_file, is_sharded)` because the serialized weights are in "pytorch format" and therefore can't be loaded using `cls.load_flax_weights`?

Reply:
Yes, that's correct! This way we call `load_pytorch_checkpoint_in_flax_state_dict` with the safetensors file, and that method checks if the file ends with `.safetensors` to load it the pytorch-way

Reply:
Thanks for explaining!